### PR TITLE
add roi_align

### PR DESCRIPTION
**Pull request by nemonameless**
### PR types
New features

### PR changes
OPs

### Describe
Add new detection api paddle.vision.ops.roi_align and paddle.vision.ops.RoIAlign

CN_DOC of paddle.vision.ops.roi_align：
https://github.com/PaddlePaddle/docs/pull/3902

image
image
image

EN_DOC of paddle.vision.ops.roi_align：
image
image
image
image

### CODE REVIEW

File: python/paddle/vision/ops.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 from ..fluid.layers import nn, utils
 from ..nn import Layer
 from ..fluid.initializer import Normal
+from ..fluid.framework import in_dygraph_mode
```
Comment (qingqing01) [inline python/paddle/vision/ops.py]: 下面 `from paddle.common_ops_import import *`  里已经将in_dygraph_mode导入了

File: python/paddle/vision/ops.py [hunk 2]
```diff
@@ -900,3 +903,164 @@ def decode_jpeg(x, mode='unchanged', name=None):
...
+    Args:
+        x (Tensor): Input feature, 4D-Tensor with the shape of [N,C,H,W], 
+            where N is the batch size, C is the input channel, H is Height,
+            W is weight. The data type is float32 or float64.
+        boxes (Tensor): Boxes (RoIs, Regions of Interest) to pool over. It 
+            should be a 2-D Tensor or 2-D LoDTensor of shape (num_boxes, 4),
+            the lod level is 1. The data type is float32 or float64. Given as
+            [[x1, y1, x2, y2], ...], (x1, y1) is the top left coordinates, and
+            (x2, y2) is the bottom right coordinates.
+        boxes_num (Tensor): The number of boxes contained in each picture in
```
Comment (qingqing01) [inline python/paddle/vision/ops.py:1174]: boxes_num需指名dtype类型

File: python/paddle/vision/ops.py [hunk 3]
```diff
@@ -900,3 +903,164 @@ def decode_jpeg(x, mode='unchanged', name=None):
...
+    In each ROI bin, the value of the four regularly sampled locations are
+    computed directly through bilinear interpolation. The output is the mean of
+    four locations. Thus avoid the misaligned problem. 
+
+    Args:
+        x (Tensor): Input feature, 4D-Tensor with the shape of [N,C,H,W], 
+            where N is the batch size, C is the input channel, H is Height,
+            W is weight. The data type is float32 or float64.
+        boxes (Tensor): Boxes (RoIs, Regions of Interest) to pool over. It 
+            should be a 2-D Tensor or 2-D LoDTensor of shape (num_boxes, 4),
```
Comment (qingqing01) [inline python/paddle/vision/ops.py]: remove 'or 2-D LoDTensor'

File: python/paddle/vision/ops.py [hunk 4]
```diff
@@ -900,3 +903,164 @@ def decode_jpeg(x, mode='unchanged', name=None):
...
+            the lod level is 1. The data type is float32 or float64. Given as
```
Comment (qingqing01) [inline python/paddle/vision/ops.py]: remove  'lod' related desc

File: python/paddle/vision/ops.py (hunk 2 above)
Comment (nemonameless) [reply python/paddle/vision/ops.py:1174]: done

**Review by qingqing01 — DISMISSED** [review]

**Review by TCChenlong — DISMISSED** [review]

**Review by TCChenlong — DISMISSED** [review]

**Review by Superjomn — DISMISSED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]

**Review by TCChenlong — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**